PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterPlayerScripts/ControlScript.rbxmx

```diff
@@ -686,6 +717,10 @@ UserInputService.GamepadConnected:connect(function(gamepadEnum)
 	ControlState:SwitchTo(ControlModules.Gamepad)
 end)
 
+MasterControl.ControlStepped:connect(function() -- Must be done constantly because there is no Humanoid.StateEnabledChanged event
+	ControlModules.Touch:EvaluateJumpButton()
```

Comment:
Can you work with Ben on this one? Rather than connecting checks to run every step we could add a Changed event for you.